File: source/HttpInterface/Debug/WaveDebugServerPage.cpp
/***************************************************************************
 *   Copyright (C) 2005-2011 Vidyasagara Guntaka                           *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "HttpInterface/Debug/WaveDebugServerPage.h"
#include "HttpInterface/Debug/Trace/WaveTraceServerMultiPage.h"

namespace WaveNs
{

WaveDebugServerPage::WaveDebugServerPage (HttpInterfaceReceiverObjectManager* pHttpInterfaceReceiverObjectManager, const string& path)
    : WaveServerPage(pHttpInterfaceReceiverObjectManager, path)
{
    m_pWaveTraceServerMultiPage = new WaveTraceServerMultiPage (pHttpInterfaceReceiverObjectManager, "/debug/trace");

    prismAssert (NULL != m_pWaveTraceServerMultiPage, __FILE__, __LINE__);
}

WaveDebugServerPage::~WaveDebugServerPage ()
{
    if (NULL != m_pWaveTraceServerMultiPage)
    {
        delete m_pWaveTraceServerMultiPage;
    }
}

}

File: Java/source/com/CxWave/Wave/HttpInterface/Debug/WaveDebugServerPage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.HttpInterface.Debug;

import com.CxWave.Wave.HttpInterface.HttpInterfaceReceiverObjectManager;
import com.CxWave.Wave.HttpInterface.WaveServerPage;
import com.CxWave.Wave.HttpInterface.Debug.Trace.WaveTraceServerMultiPage;

public class WaveDebugServerPage extends WaveServerPage
{
    public WaveDebugServerPage (final HttpInterfaceReceiverObjectManager httpInterfaceReceiverObjectManager, final String path)
    {
        super (httpInterfaceReceiverObjectManager, path);

        final WaveTraceServerMultiPage waveTraceServerMultiPage = new WaveTraceServerMultiPage (httpInterfaceReceiverObjectManager, "/debug/*/trace");
    }

}
